crates/lib3h/src/message_encoding.rs

```diff
@@ -42,15 +42,27 @@ pub mod encoding_protocol {
 
     #[derive(Debug)]
     pub enum RequestToChild {
-        Decode { payload: Opaque },
-        EncodeHandshake { space_address: String, id: String },
-        EncodePayload { payload: Opaque },
+        Decode {
+            payload: Opaque,
+        },
+        EncodeHandshake {
+            network_or_space_address: Address,
+            id: String,
```

Comment:
I'm totally cool with this as a merge-able first cut. But these are either network_hash + node_pubkey or space_hash + agent_pubkey. I'm not sure the best way to handle that for clarity.

Reply:
yeah, not ideal, but a step better than having:
`space_address: self.config.network_id.id.clone()`